core/deployment/src/test/java/io/quarkiverse/langchain4j/test/guardrails/InputGuardrailPromptTemplateTest.java

```diff
@@ -0,0 +1,186 @@
+import io.quarkiverse.langchain4j.guardrails.InputGuardrailResult;
+import io.quarkiverse.langchain4j.guardrails.InputGuardrails;
+import io.quarkiverse.langchain4j.runtime.aiservice.NoopChatMemory;
+import io.quarkus.arc.Arc;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class InputGuardrailPromptTemplateTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(MyAiService.class, MyAiService.class, GuardrailValidation.class,
+                            MyChatModel.class, MyChatModelSupplier.class, MyMemoryProviderSupplier.class));
+    @Inject
+    MyAiService aiService;
+
+    @Inject
+    GuardrailValidation guardrailValidation;
+
+    @BeforeEach
+    void setup() {
+        Arc.container().requestContext().activate();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Arc.container().requestContext().deactivate();
+    }
+
+    @Test
+    void shouldWorkNoParameters() {
+        aiService.getJoke();
+        assertThat(guardrailValidation.spyPromptTemplate()).isEqualTo("Tell me a joke");
+        assertThat(guardrailValidation.spyVariables()).isEmpty();
+    }
+
+    @Test
+    void shouldWorkWithMemoryId() {
+        aiService.getAnotherJoke("memory-id-001");
+        assertThat(guardrailValidation.spyPromptTemplate()).isEqualTo("Tell me another joke");
+        assertThat(guardrailValidation.spyVariables()).containsExactlyInAnyOrderEntriesOf(Map.of(
+                "memoryId", "memory-id-001",
+                "it", "memory-id-001" // is this correct?
+        ));
+    }
+
+    @Test
+    void shouldWorkWithNoMemoryIdAndOneParameter() {
+        aiService.sayHiToMyFriendNoMemory("Rambo");
+        assertThat(guardrailValidation.spyPromptTemplate()).isEqualTo("Say hi to my friend {friend}!");
+        assertThat(guardrailValidation.spyVariables())
+                .containsExactlyInAnyOrderEntriesOf(Map.of(
+                        "friend", "Rambo",
+                        "it", "Rambo"));
+    }
+
+    @Test
+    void shouldWorkWithMemoryIdAndOneParameter() {
+        aiService.sayHiToMyFriend("1", "Chuck Norris");
+        assertThat(guardrailValidation.spyPromptTemplate()).isEqualTo("Say hi to my friend {friend}!");
+        assertThat(guardrailValidation.spyVariables())
+                .containsExactlyInAnyOrderEntriesOf(Map.of(
+                        "friend", "Chuck Norris",
+                        "mem", "1"));
+    }
+
+    @Test
+    void shouldWorkWithNoMemoryIdAndThreeParameters() {
+        aiService.sayHiToMyFriends("Chuck Norris", "Jean-Claude Van Damme", "Silvester Stallone");
+        assertThat(guardrailValidation.spyPromptTemplate()).isEqualTo("Tell me something about {topic1}, {topic2}, {topic3}!");
+        assertThat(guardrailValidation.spyVariables())
+                .containsExactlyInAnyOrderEntriesOf(Map.of(
```

Comment:
What if the method receives a `List<String>`  ?

Reply:
I have included more tests to show how it behaves.